finality-aleph/src/data_io/legacy/data_provider.rs

```diff
@@ -400,11 +400,29 @@ mod tests {
         )
     }
 
-    // Sleep enough time so that the internal refreshing in ChainTracker has time to finish.
+    // Even though sleep time should be more enough, this may return None instead of Some very occasionally.
     async fn sleep_enough() {
         sleep(REFRESH_INTERVAL + REFRESH_INTERVAL + REFRESH_INTERVAL).await;
     }
 
+    // Retries `sleep_enough` until data in provider is available.
+    // This theoretically might fail, but practically shouldn't.
+    async fn sleep_until_data_available(data_provider: &mut DataProvider) -> AlephData {
```

Comment:
I don't like that we have to repeat exactly the same code for the legacy version. There's nothing we can do :fr: with it in this PR I guess, but maybe (the team) should consider at least having some `common` files for legacy and current in the future. (just a comment so resolve it)

Reply:
Even better, we'll be removing whole legacy in data_io after release 13 - A0-3461